internal/pkg/drand/drand_grpc.go

```diff
@@ -59,17 +82,30 @@ func (d *GRPC) ReadEntry(ctx context.Context, drandRound Round) (*Entry, error)
 			continue
 		}
 
-		return &Entry{
+		entry := &Entry{
 			Round: drandRound,
 			Signature: crypto.Signature{
 				Type: crypto.SigTypeBLS,
 				Data: pub.GetSignature(),
 			},
-		}, nil
+			parentRound: Round(pub.PreviousRound),
+		}
+		d.updateLocalState(entry)
+		return entry, nil
 	}
 	return nil, errors.New("Could not retrieve drand randomess from any address")
 }
 
+func (d *GRPC) updateLocalState(entry *Entry) {
+	if d.latestEntry == nil {
+		d.latestEntry = entry
+	}
+	if entry.Round > d.latestEntry.Round {
+		d.latestEntry = entry
+	}
+	d.cache[entry.Round] = entry
```

Comment:
This leaks. An LRU read-through cache should be really effective here. Maybe not the best time to deal with it, but an issue and TODO wouuld be good.